forge-gui/res/cardsfolder/upcoming/mystery_key.txt

```diff
@@ -0,0 +1,9 @@
+Name:Mystery Key
+ManaCost:1 U
+Types:Artifact Equipment
+T:Mode$ DamageDone | ValidSource$ Creature.EquippedBy | ValidTarget$ Player | CombatDamage$ True | Execute$ TrigSacrifice | TriggerZones$ Battlefield | TriggerDescription$ When equipped creature deals combat damage to a player, sacrifice CARDNAME. If you do, draw three cards.
+SVar:TrigSacrifice:DB$ SacrificeAll | ValidCards$ Card.Self | SubAbility$ DBDraw
```

Comment:
Probably no reason to do this with the All-Variant
But more importantly need to handle the "If you do" part in case this trigger is on the stack more than once

Reply:
I don't quite understand, there's another version like this:
```
Name:Mystery Key
ManaCost:1 U
Types:Artifact Equipment
T:Mode$ DamageDone | ValidSource$ Creature.EquippedBy | ValidTarget$ Player | CombatDamage$ True | Execute$ TrigSacrifice | TriggerZones$ Battlefield | TriggerDescription$ When equipped creature deals combat damage to a player, sacrifice CARDNAME. If you do, draw three cards.
SVar:TrigSacrifice:AB$ Draw | NumCards$ 3 | Cost$ Mandatory Sac<1/CARDNAME> | SpellDescription$ If you do, draw three cards.
SVar:NonStackingAttachEffect:True
K:Equip:1
Oracle:When equipped creature deals combat damage to a player, sacrifice Mystery Key. If you do, draw three cards.\nEquip {1} ({1}: Attach to target creature you control. Equip only as a sorcery.)
```
But I don't know which is more correct. Does the draw 3 cards is put on the stack?

Reply:
Yea that one would be fine

Only "when you do" is reflexive trigger

Reply:
The SpellDescription$ will probably not be needed again?

Reply:
done